vendor/vendor-android/base/src/main/kotlin/com/malinskiy/marathon/android/executor/logcat/LogcatCollector.kt

```diff
@@ -0,0 +1,84 @@
+package com.malinskiy.marathon.android.executor.logcat
+
+import com.malinskiy.marathon.android.executor.logcat.BatchLogSaver.SaveEntry
+import com.malinskiy.marathon.android.executor.logcat.model.LogcatEvent
+import com.malinskiy.marathon.android.executor.logcat.parse.LogcatEventsListener
+import com.malinskiy.marathon.device.Device
+import com.malinskiy.marathon.report.logs.LogTest
+import com.malinskiy.marathon.log.MarathonLogging
+import com.malinskiy.marathon.report.logs.LogReport
+import com.malinskiy.marathon.report.logs.LogReportProvider
+
+class LogcatCollector : LogcatEventsListener, LogReportProvider {
+
+    private val logger = MarathonLogging.logger(LogcatCollector::class.java.simpleName)
+
+    private val devices: MutableMap<Device, DeviceState> = hashMapOf()
+    private val batchCollectors: MutableMap<String, BatchLogSaver> = hashMapOf()
+
+    @Synchronized
```

Comment:
this method is invoked on "Tailer" thread (unique thread for each device that monitors logcat and invokes callback for each logcat message) and on "BootWait" thread pool